### PR TITLE
New. Form decoration. `Holiday form decoration` option added.

### PR DESCRIPTION
**Pull request by Glomberg**
https://doboard.com/task/8294

### CODE REVIEW

**Review by svfcode — APPROVED** [review]

File: lib/Cleantalk/ApbctWP/FormDecorator/DecorationSet.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                %HEADER_SVG%<span id="%HEADER_TEXT_ID%" class="%HEADER_TEXT_CLASS%">%HEADER_TEXT%</span>
+            </div>
+        </div>
+        ';
+
+        $template = str_replace('%HEADER_WRAPPER_ID%', $prefix . 'wrapper', $template);
+        $template = str_replace('%HEADER_WRAPPER_CLASS%', $prefix . 'wrapper', $template);
+        $template = str_replace('%HEADER_CONTENT_ID%', $prefix . 'content', $template);
+        $template = str_replace('%HEADER_CONTENT_CLASS%', $prefix . 'content', $template);
+        //    display: flex;
```
Comment (svfcode) [inline lib/Cleantalk/ApbctWP/FormDecorator/DecorationSet.php:82]: Remove comments please.

**Review by svfcode — CHANGES_REQUESTED** [review]

File: lib/Cleantalk/ApbctWP/FormDecorator/DecorationSet.php (hunk 1 above)
Comment (AntonV1211) [reply lib/Cleantalk/ApbctWP/FormDecorator/DecorationSet.php]: Ok

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 2]
```diff
@@ -311,11 +314,11 @@ private function process($arr)
                     if (empty($this->processed_data['email'])) {
                         // if found new very first email field, set it as the processed email field.
                         $this->processed_data['email'] = $value_for_email;
-                        $this->processed_data['emails_array'][$this->prev_name . $key] = $value_for_email;
+                        $this->processed_data['emails_array'][$nested_array_key] = $value_for_email;
                     } else {
                         // if processed one is already exists, set it to the message field.
-                        $this->processed_data['message'][$this->prev_name . $key] = $value_for_email;
-                        $this->processed_data['emails_array'][$this->prev_name . $key] = $value_for_email;
+                        $this->processed_data['message'][$nested_array_key] = $value_for_email;
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:320]: Please, revive this from dev branch

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 3]
```diff
@@ -311,11 +314,11 @@ private function process($arr)
...
+                        $this->processed_data['emails_array'][$nested_array_key] = $value_for_email;
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:321]: Please, revive this from dev branch

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 4]
```diff
@@ -342,14 +345,14 @@ private function process($arr)
                     } elseif (count($match_nickname) > 1) {
                         $this->processed_data['nickname']['nick'] = $value;
                     } else {
-                        $this->processed_data['message'][$this->prev_name . $key] = $value;
+                        $this->processed_data['message'][$nested_array_key] = $value;
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:348]: Please, revive this from dev branch

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 5]
```diff
@@ -342,14 +345,14 @@ private function process($arr)
...
                     }
                     // Subject
                 } elseif ($this->processed_data['subject'] === '' && false !== stripos($key, "subject")) {
                     $this->processed_data['subject'] = $value;
                     // Message
                 } else {
-                    $this->processed_data['message'][$this->prev_name . $key] = $value;
+                    $this->processed_data['message'][$nested_array_key] = $value;
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:355]: Please, revive this from dev branch

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 6]
```diff
@@ -360,7 +363,7 @@ private function process($arr)
                 }
 
                 $prev_name_original = $this->prev_name;
-                $this->prev_name    = ($this->prev_name === '' ? $key . '_' : $this->prev_name . $key . '_');
+                $this->prev_name    = ($this->prev_name === '' ? $key : $this->prev_name . $key);
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:366]: Please, revive this from dev branch

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 7]
```diff
@@ -574,8 +577,21 @@ private static function getAvailablePOSTFieldsForCurrentRequest()
...
+            if (is_array($value)) {
+                foreach ($value as $nested_key => $nested_value) {
+                    // @todo Make recursive converting here
+                    $fields[$key . '[' . $nested_key . ']'] = $nested_value;
+                }
+            } else {
+                $fields[$key] = $value;
+            }
+        }
+
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:592]: Please, revive this from dev branch

File: lib/Cleantalk/ApbctWP/GetFieldsAny.php [hunk 8]
```diff
@@ -574,8 +577,21 @@ private static function getAvailablePOSTFieldsForCurrentRequest()
...
         // @ToDo we have to implement a logic to find form fields (fields names, fields count) in serialized/nested/encoded items. not only $_POST.
-        return $_POST;
+        return $fields;
```
Comment (alexandergull) [inline lib/Cleantalk/ApbctWP/GetFieldsAny.php:594]: Please, revive this from dev branch

**Review by svfcode — APPROVED** [review]

**Review by alexandergull — APPROVED** [review]

**Pull request closed without merging**